### PR TITLE
Add ufunc methods

### PR DESCRIPTION
**Pull request by bryevdv**

### CODE REVIEW

File: cunumeric/coverage.py [hunk 1]
```diff
@@ -250,6 +250,37 @@ def clone_module(
                 cast(AnyCallable, value), mod_name, attr, reporting=reporting
             )
             new_globals[attr] = wrapped
+            if isinstance(value, lgufunc):
+                wrapped.reduce = implemented(  # type: ignore
```
Comment (manopapad) [inline cunumeric/coverage.py:254]: Wrapping these as implemented means we don't get the performance warning when falling back. I would expect something like this:
```
print(cn.add.accumulate(cn.arange(8)))
```
to print out the "cuNumeric has not implemented ... and is falling back" warning. We could wrap them as unimplemented here, or maybe on each individual method in ufunc.py.

Comment (bryevdv) [reply cunumeric/coverage.py:254]: typo, should have been `unimplemented`

Comment (bryevdv) [reply cunumeric/coverage.py:254]: actually, `reduce` is implemented, isn't it (it is the only one, AFAICT) `reduce` specifically does
```
        array = convert_to_cunumeric_ndarray(array)

        if self._red_code is None:
            raise NotImplementedError(
                f"reduction for {self} is not yet implemented"
            )
        if not isinstance(where, bool) or not where:
            raise NotImplementedError(
                "the 'where' keyword is not yet supported"
            )

        # NumPy seems to be using None as the default axis value for scalars
        if array.ndim == 0 and axis == 0:
            axis = None

        # TODO: Unary reductions still need to be refactored
        return array._perform_unary_reduction(
            self._red_code,
            array,
            axis=axis,
            dtype=dtype,
            out=out,
            keepdims=keepdims,
            initial=initial,
            where=where,
        )
```

Comment (bryevdv) [reply cunumeric/coverage.py:254]: marked all as `unimplemented` except `reduce` in https://github.com/nv-legate/cunumeric/pull/834/commits/de80e2a0786dc3141cf2ef525d893ab92a9b7997

**Comment by manopapad** [comment]
Actually I think the fallback can happen by `unimplemented` directly, without adding custom code to ufunc. I pushed a change to do this that seems to work, but feel free to revert if you don't like it.

File: cunumeric/_ufunc/ufunc.py [hunk 2]
```diff
@@ -186,6 +186,10 @@ def __init__(self, name: str, doc: str) -> None:
         self._name = name
         self.__doc__ = doc
 
+    @property
+    def __name__(self) -> str:
```
Comment (manopapad) [inline cunumeric/_ufunc/ufunc.py:190]: Actually this isn't needed, but it's presumably part of the documented ufunc API, so it doesn't hurt (?)

File: cunumeric/coverage.py [hunk 3]
```diff
@@ -251,36 +253,24 @@ def clone_module(
...
+                    wrapped_method = (
+                        implemented(
+                            getattr(value, method),
+                            f"{mod_name}.{attr}",
+                            method,
+                            reporting=reporting,
+                        )
+                        if hasattr(value, method)
+                        else unimplemented(
+                            getattr(getattr(origin_module, attr), method),
```
[the remark this answers is not in the document]
Comment (bryevdv) [reply cunumeric/coverage.py:266]: Actually I guess we don't really have any tests for functions that are *not* implemented. I did check manually that everything works as expected. Are there any other changes to make?

File: cunumeric/_ufunc/ufunc.py (hunk 2 above)
Comment (bryevdv) [reply cunumeric/_ufunc/ufunc.py:190]: I would expect it
```

In [1]: import numpy as np

In [2]: np.add
Out[2]: <ufunc 'add'>

In [3]: np.add.__name__
Out[3]: 'add'
```

File: cunumeric/coverage.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (manopapad) [reply cunumeric/coverage.py:266]: We could add a test to exercise the ufunc methods on a binary and a unary ufunc.

[the remark this answers is not in the document]
Comment (bryevdv) [reply cunumeric/coverage.py:266]: https://github.com/nv-legate/cunumeric/pull/834/commits/c6ba99d48c1a9cb5821ef1e953090fa5976a7007

**Review by manopapad — APPROVED** [review]

**Pull request merged**